packages/core/src/external/carequality/ihe-gateway-v2/outbound/__tests__/process-xcpd.test.ts

```diff
@@ -126,3 +129,97 @@ describe("processXCPDResponse", () => {
     expect(response.patientMatch).toBeNull();
   });
 });
+
+describe("convertIheAddressToAddress", () => {
+  it("should correct invalid streetAddressLine", () => {
+    const iheAddresses = {
+      streetAddressLine: [2831],
+      city: "San Francisco",
+      state: "CA",
+      postalCode: "94101",
+      country: "USA",
+    };
+    const result = convertIheAddressToAddress(iheAddresses);
+    expect(result).toEqual({
+      line: ["2831"],
+      city: "San Francisco",
+      country: "USA",
+      postalCode: "94101",
+      state: "CA",
+    });
+    if (result) {
+      const isValid = isValidAddress(result);
+      expect(isValid).toBeTruthy();
+    } else {
+      throw new Error("Result is null");
+    }
+  });
+  it("should filter out empty address", () => {
+    const iheAddresses = {
+      streetAddressLine: [],
+      city: "San Francisco",
+      state: "CA",
+      postalCode: "94101",
+      country: "USA",
+    };
+    const result = convertIheAddressToAddress(iheAddresses);
+    expect(result).toEqual({
+      city: "San Francisco",
+      state: "CA",
+      postalCode: "94101",
+      country: "USA",
+    });
+    if (result) {
+      const isValid = isValidAddress(result);
+      expect(isValid).toBeTruthy();
+    } else {
+      throw new Error("Result is null");
+    }
+  });
+  it("should filter out 0 as city value", () => {
+    const iheAddresses = {
+      streetAddressLine: "111 Market St",
+      city: 0,
+      state: "CA",
+      postalCode: "94101",
+      country: "USA",
+    };
+    const result = convertIheAddressToAddress(iheAddresses);
+    expect(result).toEqual({
+      line: ["111 Market St"],
+      state: "CA",
+      postalCode: "94101",
+      country: "USA",
+    });
+
+    if (result) {
+      const isValid = isValidAddress(result);
+      expect(isValid).toBeTruthy();
+    } else {
+      throw new Error("Result is null");
+    }
+  });
+  it("should stringify 1 as city value", () => {
```

Comment:
What's the reason to accept non-zero integers as city but not zero?

I understand the code that generates this, but the tests are not just to make the code pass, so I'm wondering why is city `1` valid by but city `0` is not.

_(I think we should not accept any numeric names)_